relayer/chain/chain.go

```diff
@@ -14,8 +14,15 @@ type Message struct {
 	Payload interface{}
 }
 
+type Header struct {
+	HeaderData interface{}
+	ProofData  interface{}
+}
+
 type Chain interface {
 	Name() string
 	Start(ctx context.Context, eg *errgroup.Group) error
 	Stop()
+	WithReceiver(messages chan Message, headers chan Header) error
```

Comment:
Would prefer to name these `SetReceiver` and `SetSender` respectively since they mutate state. In my view, `With` is often used in constructor/factory method names when an object is being created with some input state.

Reply:
Good catch :) This started off as factory methods until I realized I want it to return an error